src/js/view/view.js

```diff
@@ -608,8 +608,11 @@ function View(_api, _model) {
             _controls.unmuteAutoplay(_api, model);
         }
 
+        const AMPController = _model.get('ampController');
+        const AMPControlsFullscreen = AMPController && !OS.iOS && !Browser.safari;
+
         if (fullscreenHelpers.supportsDomFullscreen()) {
-            if (state) {
+            if (state && !AMPControlsFullscreen) {
```

Comment:
If amp controller is a commercial concept then you should create a commercial `fullscreenHelpers` module to handle this logic instead of introducing amp logic here.

Reply:
kk, changing it over now. Originally I wanted to avoid that because that would be a much larger open source change, since those functions are constants scoped in view.init and less DRY (it's the same function but we check if it's AMP and not iOS/Safari before delegating the exit/request to AMP). I'll move them now.